api/api/routers/tools_router.py

```diff
@@ -0,0 +1,56 @@
+from fastapi import APIRouter, HTTPException
+
+from api.dependencies.services import ToolsServiceDep
+from api.dependencies.task_info import TaskInfoDep
+from api.routers.view_models.tools_view_models import ToolParamsRequest, ToolResponse
+from core.storage import ObjectNotFoundException
+
+router = APIRouter(prefix="/agents/{task_id}/tools")
+
+# The endpoint is this router have a 'task' dependency that just checks that the task in url exists but the task is not used since we fetch all the tools from the tenant.
+
+
+@router.get("")
+async def list_tools(task: TaskInfoDep, tools_service: ToolsServiceDep) -> list[ToolResponse]:
+    tools = await tools_service.list_tools()
+    return [ToolResponse.from_domain(tool) for tool in tools]
+
+
+@router.get("/{id}")
+async def get_tool(id: str, task: TaskInfoDep, tools_service: ToolsServiceDep) -> ToolResponse:
```

Comment:
task is unused here as well ? (but could remove)

Reply:
see https://github.com/WorkflowAI/WorkflowAI/pull/204/files#r2060332246